### ISSUE TITLE
considering `pad_to_multiple_of` for run_mlm.py

### ISSUE THREAD

**Issue by dorost1234** [huggingface/transformers#10627]
## Environment info

- `transformers` version: 4.3.3
- Platform: linux
- Python version: 3.7
- PyTorch version (GPU?): 1.8
- Tensorflow version (GPU?): - 
- Using GPU in script?: - 
- Using distributed or parallel set-up in script?: - 

### Who can help
@sgugger, @patil-suraj

## Information
I have seen in huggingface codes such as run_seq2seq.py that for the case of fp16, they pad to multiple of 8 like below, perhpas for efficiency purpose:

```
 data_collator = DataCollatorForSeq2Seq(
        tokenizer,
        label_pad_token_id=label_pad_token_id,
        pad_to_multiple_of=8 if training_args.fp16 else None,
       )
```

For run_mlm.py codes, the padding condition in case of fp16 is not considered, could it be a bug and possibly performance could get better in run_mlm.py if this was set? kindly appreciate  having a look.

thanks

**Comment by sgugger** [comment]
Indeed, we could use that when the line by line option is set (otherwise there is just no padding). Would you like to make a PR with this?
